components/Buttons/BackToTopButton.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useState, useEffect } from 'react';
+
+const BackToTopButton: React.FC = () => {
+  const [isVisible, setIsVisible] = useState(false);
+
+  // Show button when page is scrolled down
+  const toggleVisibility = () => {
```

Comment:
Running this function on every scroll event isn’t ideal; it creates a performance bottleneck, especially on mobile. It’d be better to throttle or debounce it to around 100–200ms. Right now, it can fire 60+ times per second 🤔